lib/seahorn/HornifyFunction.cc

```diff
@@ -137,13 +142,64 @@ void HornifyFunction::extractFunctionInfo(const BasicBlock &BB) {
   m_db.addRule(allVars, bind::fapp(fi.sumPred, postArgs));
 
   // -- expose basic properties of the summary
-  postArgs[0] = bind::boolConst(mkTerm(std::string("arg.0"), m_efac));
-  postArgs[1] = bind::boolConst(mkTerm(std::string("arg.1"), m_efac));
-  postArgs[2] = bind::boolConst(mkTerm(std::string("arg.2"), m_efac));
-  m_db.addConstraint(
-      bind::fapp(fi.sumPred, postArgs),
-      mk<AND>(mk<OR>(postArgs[0], mk<EQ>(postArgs[1], postArgs[2])),
-              mk<OR>(mk<NEG>(postArgs[0]), mk<NEG>(postArgs[1]), postArgs[2])));
+  // For partial function stubs, a summary must be synthesized.
+  if (!fi.isInferable) {
+    postArgs[0] = bind::boolConst(mkTerm(std::string("arg.0"), m_efac));
+    postArgs[1] = bind::boolConst(mkTerm(std::string("arg.1"), m_efac));
+    postArgs[2] = bind::boolConst(mkTerm(std::string("arg.2"), m_efac));
+    m_db.addConstraint(
+        bind::fapp(fi.sumPred, postArgs),
+        mk<AND>(
+            mk<OR>(postArgs[0], mk<EQ>(postArgs[1], postArgs[2])),
+            mk<OR>(mk<NEG>(postArgs[0]), mk<NEG>(postArgs[1]), postArgs[2])));
+  }
+}
+
+std::list<llvm::Value *> HornifyFunction::getPartialFnsToSynth(BasicBlock &BB) {
+  // Gets reference to sea.synth.assert.
+  if (!m_synthAssertFn) {
+    auto &SBI = m_parent.getSBI();
+    auto &M = (*BB.getParent()->getParent());
+    m_synthAssertFn = SBI.mkSeaBuiltinFn(SeaBuiltinsOp::SYNTH_ASSERT, M);
+  }
+
+  // Records all functions passed to sea.synth.assert within BB.
+  std::list<llvm::Value *> partials;
+  for (auto &I : BB) {
+    // Must be a call to sea.synth.assert.
+    if (!isa<CallInst>(&I))
+      continue;
+
+    // Analyzes the call.
+    CallSite CS(&I);
+    if (const Function *fn = CS.getCalledFunction()) {
+      if (fn->getName().equals(m_synthAssertFn->getName())) {
```

Comment:
instead of iterating over all instructions, iterate over all uses of `m_synthAssertFn`